homeassistant/components/overkiz/update.py

```diff
@@ -0,0 +1,113 @@
+"""Support for Overkiz updates."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyoverkiz.enums import UIClass, UIWidget, UpdateBoxStatus
+from pyoverkiz.models import Gateway
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantOverkizData
+from .const import DOMAIN
+from .entity import OverkizEntity
+
+UPDATE_COMMAND = "update"
+
+UPDATE_AVAILABLE_STATUS = {
+    UpdateBoxStatus.READY_TO_UPDATE,
+    UpdateBoxStatus.READY_TO_BE_UPDATED_BY_SERVER,
+    UpdateBoxStatus.READY_TO_UPDATE_LOCALLY,
+    UpdateBoxStatus.READY_TO_UPDATE_LOCALLY,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Overkiz update from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+
+    entities: list[OverkizUpdateEntity] = []
+
+    for device in data.platforms[Platform.UPDATE]:
```

Comment:
Perhaps we should just loop here over all devices and check for widget or UI (why both?). Or just check for update command, so it can work with other update command as well in the future.